fs/btrfs/zstd.c

```diff
@@ -95,7 +114,26 @@ static int zstd_compress_pages(struct list_head *ws,
 	unsigned long len = *total_out;
 	const unsigned long nr_dest_pages = *out_pages;
 	unsigned long max_out = nr_dest_pages * PAGE_SIZE;
-	ZSTD_parameters params = zstd_get_btrfs_parameters(len);
+
+	ZSTD_parameters params = zstd_get_btrfs_parameters(len, workspace->level);
+	size_t new_size = ZSTD_CStreamWorkspaceBound(params.cParams);
```

Comment:
nit: The kernel style puts all variable declarations without spaces, then a single space.

`new_size` can be const.